nncf/torch/quantization/layers.py

```diff
@@ -744,12 +768,15 @@ def signed(self, signed: bool):
         self.set_levels()
 
     def quantize(self, x, execute_traced_op_as_identity: bool = False):
+        with DisableTorchFunction():
+            # in multi-device case after loading nncf checkpoint, quantizers have a different device.
+            self.to(x.device)
```

Comment:
Is `self.to(x.device)` called in forward? If so, this seems like a workaround. Is there a way to handle this when loading nncf checkpoint?

Reply:
I reverted this change for now, since it's about corner case on multi-device tuning and loading checkpoint.
I was planning to add a PR with sample and checkpoint loading functionality, can cover this there.

Reply:
Some matching of `compression_module` from insertion point to the corresponding target module is needed in order to align devices between them.

These matching looks flimsy:
https://github.com/openvinotoolkit/nncf/pull/3341/files#diff-bfabae4564efa2b64a87b5c99178c9ad75a1eadf5e4241def5c09b76d440bb05R81-R101

Did you have any other idea in mind?